### PR TITLE
Add DeleteRange and go.mod support

### PR DESCRIPTION
**Pull request by driver-devel**
## Notes

- Cherry pick commits related to DeleteRange from upstream
- Add go mod support
- Add tests in docker using docker compose

### CODE REVIEW

File: test.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# This dockerile might be helpful for running gorocksdb tests in docker
```
Comment (ferhatelmas) [inline test.Dockerfile]: Add it into makefile maybe, later we can setup CI

File: test.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+RUN --mount=type=ssh git clone git@github.com:GetStream/rocksdb.git /rocksdb
+RUN cd /rocksdb && \
+    git checkout broadwell && \
+    DISABLE_JEMALLOC=1 make static_lib -j5
+
+RUN go get github.com/facebookgo/ensure && \
+    go get github.com/stretchr/testify/assert
+
+ENV CGO_CFLAGS="-I/rocksdb/include"
+ENV CGO_LDFLAGS="-L/rocksdb -lrocksdb -lstdc++ -lm -lz -lbz2 -lsnappy -llz4 -lzstd"
```
Comment (ferhatelmas) [inline test.Dockerfile]: FYI, in the newer versions, we need `-ldl` too

**Review by ferhatelmas — APPROVED** [review]
LGTM

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.PHONY: docker-clean
+docker-clean:
+	@docker compose down -v --remove-orphans
+
+.PHONY: docker-test
+docker-test:
+	@docker compose build test && docker compose run --rm test go test -race=1 -v ./...
```
Comment (ferhatelmas) [inline Makefile]: Maybe add `GODEBUG=cgocheck=2` for cgo

Comment (ferhatelmas) [reply Makefile]: Are you going to add an action to run tests?

Comment (driver-devel) [reply Makefile]: We can do it, but i need to setup CI access to rocksdb repo then

Comment (ferhatelmas) [reply Makefile]: We can do it separately.

**Pull request merged**